pysrc/juliacall/__init__.py

```diff
@@ -248,8 +248,37 @@ def jlstr(x):
                 "PYTHON_JULIACALL_HANDLE_SIGNALS=no."
             )
 
-init()
+    # Next, automatically load the juliacall extension if we are in a Jupyter notebook
+    CONFIG['autoload_extensions'] = choice('autoload_extensions', ['yes', 'no'])[0]
+    if CONFIG['autoload_extensions'] in {'yes', None}:
+        try:
+            get_ipython = sys.modules['IPython'].get_ipython
+
+            if 'IPKernelApp' not in get_ipython().config:
```

Comment:
What does this check do? Please add a comment.

Reply:
Done.

Btw, I learned about it from tqdm's automatic functionality: https://github.com/tqdm/tqdm/blob/master/tqdm/autonotebook.py. If Python ever updates how this works, tqdm is probably a good place to figure out what to do for this type of behavior.